src/database/src/output/airspace.rs

```diff
@@ -0,0 +1,185 @@
+    pub direction: TurnDirection,
+}
+
+#[derive(Serialize, Debug, Copy, Clone)]
+pub enum PathType {
+    #[serde(rename = "C")]
+    Circle,
+    #[serde(rename = "G")]
+    GreatCircle,
+    #[serde(rename = "R")]
+    RhumbLine,
+    #[serde(rename = "A")]
+    Arc,
+}
+
+#[serde_with::skip_serializing_none]
+#[derive(Serialize, Debug)]
+pub struct Path {
+    pub location: Coordinates,
+    pub arc: Option<Arc>,
+    pub path_type: PathType,
+}
+
+impl Path {
+    fn from_data(
+        latitude: Option<f64>, longitude: Option<f64>, arc_latitude: Option<f64>, arc_longitude: Option<f64>,
+        arc_distance: Option<f64>, arc_bearing: Option<f64>, boundary_via: String,
+    ) -> Self {
+        let boundary_char = boundary_via.chars().nth(0).unwrap();
+        match boundary_char {
+            'C' => Self {
+                location: Coordinates {
+                    lat: arc_latitude.unwrap(),
+                    long: arc_longitude.unwrap(),
+                },
+                arc: None,
+                path_type: PathType::Circle,
+            },
+            'G' | 'H' => Self {
+                location: Coordinates {
+                    lat: latitude.unwrap(),
+                    long: longitude.unwrap(),
+                },
+                arc: None,
+                path_type: match boundary_char {
+                    'G' => PathType::GreatCircle,
+                    'H' | _ => PathType::RhumbLine,
+                },
+            },
+            'L' | 'R' => Self {
+                location: Coordinates {
+                    lat: latitude.unwrap(),
+                    long: longitude.unwrap(),
+                },
+                arc: Some(Arc {
+                    origin: Coordinates {
+                        lat: arc_latitude.unwrap(),
+                        long: arc_longitude.unwrap(),
+                    },
+                    distance: arc_distance.unwrap(),
+                    bearing: arc_bearing.unwrap(),
+                    direction: match boundary_char {
+                        'R' => TurnDirection::Right,
+                        'L' | _ => TurnDirection::Left,
+                    },
+                }),
+                path_type: PathType::Arc,
+            },
+            _ => panic!("Invalid path type"),
+        }
+    }
+}
+
+#[derive(Serialize, Debug)]
+pub struct ControlledAirspace {
+    pub area_code: String,
+    pub icao_code: String,
+    pub airspace_center: String,
+    pub name: String,
+    pub airspace_type: ControlledAirspaceType,
+    pub boundary_paths: Vec<Path>,
+}
+
+#[derive(Serialize, Debug)]
+pub struct RestrictiveAirspace {
+    pub area_code: String,
+    pub icao_code: String,
+    pub designation: String,
+    pub name: String,
+    pub airspace_type: RestrictiveAirspaceType,
+    pub boundary_paths: Vec<Path>,
+}
+
+pub(crate) fn map_controlled_airspaces(data: Vec<sql_structs::ControlledAirspace>) -> Vec<ControlledAirspace> {
+    let mut airspace_complete = false;

```

Comment:
Going off of how we have already declared public functions, why is this `pub(crate)` as opposed to just `pub` as established in other files?

Reply:
Some functions we don't want accessible outside of the crate